ksql-execution/src/main/java/io/confluent/ksql/execution/builder/KsqlQueryBuilder.java

```diff
@@ -196,4 +203,22 @@ private void track(final String loggerNamePrefix, final PersistenceSchema schema
     }
     schemas.put(loggerNamePrefix, schema);
   }
+
+  public Optional<Topology.AutoOffsetReset> getAutoOffsetReset() {
```

Comment:
I'm a little confused why this belongs here?

Reply:
We're doing this weird thing in our plan build where one plan node can ask its source node to set auto.offset.reset to earliest, but it does that by setting the KsqlConfig. It would make more sense to just set a field in `KsqlQueryBuilder`. This change at least has a sane interface for controlling this.

Reply:
Yeah, I'd either implement the change fully, or revert this in this PR and do it in a follow up.